unlock-app/src/__tests__/components/interface/keyChain/QRModal.test.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import * as rtl from 'react-testing-library'
+import QRModal from '../../../../components/interface/keyChain/QRModal'
+
+let dismiss: jest.Mock<any, any>
+let sendEmail: jest.Mock<any, any>
+const render = () => {
+  dismiss = jest.fn()
+  sendEmail = jest.fn()
+  return rtl.render(
+    <QRModal
+      active
+      dismiss={dismiss}
+      sendEmail={sendEmail}
+      value="just some test data"
+    />
+  )
+}
+
+describe('QRModal', () => {
+  it('should render a QR code', () => {
+    expect.assertions(1)
+    const { container } = render()
+
+    expect(container.querySelector('canvas')).not.toBeNull()
+  })
+
+  it('should have a disabled submit button when input is empty', () => {
+    expect.assertions(1)
+    const { getByText } = render()
+
+    const submit = getByText('Send Email')
+    rtl.fireEvent.click(submit)
+
+    expect(sendEmail).not.toHaveBeenCalled()
```

Comment:
In this test is there a way to see if the button has a disabled state? From first glance it looks like the test does ensure that it will not fire off an email but I guess it may still be enabled?

Reply:
In the code, there are 2 buttons and we choose which to render based on the validity of the input field. One button is hooked up to the submit function, and one is not. So this test demonstrates (in conjunction with the other test that shows the function _does_ get called when the input is filled) which of the buttons renders when.

There is a more direct way to identify the buttons, by giving each a `data-testid` that we can query against. That's less idiomatic though.